stores/milvus-documentstore/milvus_documentstore/milvus.py

```diff
@@ -401,196 +409,294 @@ def query_by_embedding(
-            anns_field=self.embedding_field,
-            param={"metric_type": self.metric_type, **self.search_param},
-            limit=top_k,
+        if filters is not None:
+            filters = (
+                LogicalFilterClause.parse(filters).convert_to_milvus()
+                if filters
+                else None
+            )
+
+        res = self.client.search(
+            collection_name=index,
+            data=[query_emb],
+            top_k=top_k,
+            include_vectors=return_embedding,
+            partition_keys=[0],
+            filter_expression=filters,
         )
+        res_docs = []
+        for hit in res[0]:
+            hit.pop(EMPTY_FIELD)
+            doc = Document.from_dict(hit)
+            doc.score = (
+                hit.score
+                if not scale_score
+                else self.scale_to_unit_interval(hit.score, self.similarity)
+            )
+            res_docs.append(doc)
 
-        vector_ids_for_query = []
-        scores_for_vector_ids: Dict[str, float] = {}
-        for vector_id, distance in zip(search_result[0].ids, search_result[0].distances):
-            vector_ids_for_query.append(str(vector_id))
-            scores_for_vector_ids[str(vector_id)] = distance
+        return res_docs
 
-        documents = self.get_documents_by_vector_ids(vector_ids_for_query, index=index)
+    def get_label_count(
+        self, index: Optional[str] = None, headers: Optional[Dict[str, str]] = None
+    ) -> int:
+        if headers:
+            raise NotImplementedError("MilvusDocumentStore does not support headers.")
 
-        if return_embedding:
-            self._populate_embeddings_to_docs(index=index, docs=documents)
+        index = index or self.collection_name
+        if index not in self.client.list_collections():
+            self._create_collection(index)
+            return 0
 
-        for doc in documents:
-            score = scores_for_vector_ids[doc.meta["vector_id"]]
-            if scale_score:
-                score = self.scale_to_unit_interval(score, self.similarity)
-            doc.score = score
+        return len(
+            self.client.query(
+                index,
+                output_fields=[ID_FIELD],
+                partition_keys=[2],
+            )
+        )
 
-        return documents
+    def write_labels(
+        self,
+        labels: Union[List[Label], List[dict]],
+        index: Optional[str] = None,
+        headers: Optional[Dict[str, str]] = None,
+        batch_size=100,
+    ):
+        if headers:
+            raise NotImplementedError("MilvusDocumentStore does not support headers.")
+
+        index = index or self.collection_name
+        if index not in self.client.list_collections():
+            self._create_collection(index)
+
+        insert_labels = []
+
+        for label in labels:
+            if isinstance(label, Label):
+                label = label.to_json()
+                label = json.loads(label)
```

Comment:
I think this can be simplified:
```suggestion
                label = label.to_dict()
```

Reply:
to_dict() isnt recursive so Answer in the Label does not get deconstructed. to_json is recursive so it works.